debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/type/connect/ConnectStringType.java

```diff
@@ -59,33 +60,25 @@ public String getTypeName(DatabaseDialect dialect, Schema schema, boolean key) {
             }
 
             if (columnSize > 0) {
-                return dialect.getTypeName(resolvedJdbcType, Size.length(columnSize));
+                return dialect.getJdbcTypeName(resolvedJdbcType, Size.length(columnSize));
             }
-            return dialect.getTypeName(resolvedJdbcType);
+            return dialect.getJdbcTypeName(resolvedJdbcType);
         }
         else {
             final int jdbcType = hasNationalizedCharacterSet(schema) ? Types.NVARCHAR : Types.VARCHAR;
-            // The column propagation details either don't exist or did not map to a logical STRING type
-            // In this case, we apply the size defaults (no size for non-keys and max-key for keys).
-            if (key) {
-                return dialect.getTypeName(jdbcType, Size.length(getMaxSizeInKey(dialect, jdbcType)));
-            }
-            return dialect.getTypeName(jdbcType);
+            return dialect.getJdbcTypeName(jdbcType, Size.length(getMaxSizeInKey(dialect, jdbcType)));
```

Comment:
Another case where non-key columns will be scoped to smaller sizes than previously.

Reply:
key handling re-introduced. see above.